code/Helper/Entity/Producthelper.php

```diff
@@ -6,11 +6,27 @@ class Algolia_Algoliasearch_Helper_Entity_Producthelper extends Algolia_Algolias
 
     protected static $_predefinedProductAttributes = array('name', 'url_key', 'description', 'image', 'thumbnail');
 
+    protected $_productTypeMap;
+
     protected function getIndexNameSuffix()
     {
         return '_products';
     }
 
+    protected function getMappedProductType($originalType)
+    {
+        if (!isset($this->_productTypeMap[$originalType])) {
+            $mappedType = (string)Mage::app()->getConfig()->getNode('default/algoliasearch/product_map/' . $originalType);
```

Comment:
Can you encapsulate this in https://github.com/algolia/algoliasearch-magento/blob/develop/code/Helper/Config.php